cmd/install.go

```diff
@@ -137,23 +137,30 @@ func getFinalPath(path, fileName string) (string, error) {
 //TODO check if other binary has the same hash and warn about it.
 //TODO if the file is zipped, tared, whatever then extract it
 func saveToDisk(f *providers.File, path string, overwrite bool) error {
-
 	defer f.Data.Close()
 
 	var buf bytes.Buffer
 	tee := io.TeeReader(f.Data, &buf)
 
 	t, err := filetype.MatchReader(tee)
-
 	if err != nil {
 		return err
 	}
 
+	var outputFile = io.MultiReader(&buf, f.Data)
+
 	if t != matchers.TypeElf && t != matchers.TypeGz {
-		return fmt.Errorf("File type [%v] not supported", t)
-	}
+		// if its not elf or gz then its an executable text file
+		// that needs to be written as is
+		file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0766)
```

Comment:
Hmm why not using variables here and follow the current function flow execution. It's a bit hard for me to understand that this is an alternative "valid" flow that this function can have.

Reply:
Fixed!